tests/int/__snapshots__/test.test.ts.snap

```diff
@@ -0,0 +1,433 @@
+        "headers": Object {},
+        "method": "GET",
+        "path": "/api",
+      },
+      "response": Object {
+        "body": Object {
+          "numberField": 10,
+          "stringField": "text",
+        },
+        "headers": Object {},
+        "status": 200,
+      },
+    },
+  ],
+  "metadata": Object {
+    "pactSpecification": Object {
+      "version": "2.0.0",
+    },
+  },
+  "provider": Object {
+    "name": "provider-name",
+  },
+}
+`;
+
+exports[`createPacts type-analysis 1`] = `
+Object {
+  "consumer": Object {
+    "name": "consumer-name",
+  },
+  "interactions": Array [
+    Object {
+      "description": "apiFunction",
+      "request": Object {
+        "headers": Object {},
+        "method": "GET",
+        "path": "/api",
+      },
+      "response": Object {
+        "body": Object {
+          "booleanField": true,
+          "constString": "CONST",
+          "enumType": "FIRST",
+          "intersectionField": Object {
+            "fieldA": "text",
+            "fieldB": true,
+            "fieldC": 10,
+          },
+          "numberEnum": 1,
+          "numberField": 10,
+          "someObject": Object {
+            "booleanField": true,
+            "numberField": 10,
+          },
+          "someOtherObject": Object {
+            "someInnerObject": Object {
+              "someNextEnum": "FIRST",
+            },
+          },
+          "someType": Object {
+            "fieldA": "text",
+            "fieldB": true,
+          },
+          "someUnionType": "AAA",
+          "stringAliasType": "text",
+          "stringArray": Array [
+            "text",
+          ],
+          "stringField": "text",
+          "unionWithDifferentTypes": "text",
+          "unionsArray": Array [
+            "OPTION_1",
+          ],
+        },
+        "headers": Object {},
+        "matchingRules": Object {
+          "$.body": Object {
+            "match": "type",
+          },
+          "$.body.constString": Object {
+            "match": "regex",
+            "regex": "CONST",
+          },
+          "$.body.enumType": Object {
+            "match": "regex",
+            "regex": "FIRST|SECOND|THIRD|OTHER",
+          },
+          "$.body.numberEnum": Object {
+            "match": "regex",
+            "regex": "1|2|3|7|8",
+          },
+          "$.body.someOtherObject.someInnerObject.someNextEnum": Object {
+            "match": "regex",
+            "regex": "FIRST|SECOND|THIRD|OTHER",
+          },
+          "$.body.someUnionType": Object {
+            "match": "regex",
+            "regex": "AAA|BBB|100",
+          },
+          "$.body.unionsArray": Object {
+            "match": "regex",
+            "regex": "OPTION_1|OPTION_2|OPTION_3",
+          },
+        },
+        "status": 200,
+      },
+    },
+  ],
+  "metadata": Object {
+    "pactSpecification": Object {
+      "version": "2.0.0",
+    },
+  },
+  "provider": Object {
+    "name": "provider-name",
+  },
+}
+`;
```

Comment:
We're missing a test with an array of some complex, nested object returned in the response body, to verify  `matchingRules` in such a case

Reply:
I've created that type of example field.